src/App.tsx

```diff
@@ -3,9 +3,68 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
+import { getMovie } from './api';
+import { MovieData } from './types/MovieData';
+import { ResponseError } from './types/ReponseError';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState(false);
+  const [loader, setLoader] = useState(false);
+
+  const changeQuery = (text: string) => {
+    setError(false);
+    setQuery(text);
+  };
+
+  function instanceOfMovie(data: MovieData | ResponseError):
+    data is MovieData {
+    return 'imdbID' in data;
```

Comment:
Wow!! you did it)) I'd like to understand by step your production)

Reply:
i googled, how to check what interface variable has) it was the clearest way for me) I think it works like that: if **data** has interface MovieData, then it returns true (**property in data**). 

Here is about _data **is** MovieData_
https://stackoverflow.com/questions/40081332/what-does-the-is-keyword-do-in-typescript

Maybe it is some easiest way to check this, i don't know))

Reply:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards